packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -355,6 +361,55 @@ def __eq__(self, other: Any) -> bool:
         return self.payload == other.payload and self.signature == other.signature
 
 
+@dataclass(frozen=True)
+class NewBaseTxPayload(ABC, metaclass=_MetaPayload):
+    """This class represents a base class for transaction payload classes."""
+
+    sender: str
+    round_count: int = field(default=ROUND_COUNT_DEFAULT, init=False)
+
+
+@dataclass
+class NewTransaction:
+    """Class to represent a transaction for the ephemeral chain of a period."""
+
+    payload: NewBaseTxPayload
+    signature: Optional[str] = None
```

Comment:
This class I propose to treat as a dataclass as well. It can contain the immutable payload which can be signed by an agent. Here we can move all operations related to serialization, deserialization and verification of the payload.